libs/fileset/glob.go

```diff
@@ -0,0 +1,45 @@
+package fileset
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+)
+
+type GlobSet struct {
+	root     string
+	patterns []string
+}
+
+func NewGlobSet(root string, includes []string) *GlobSet {
+	return &GlobSet{root, includes}
+}
+
+// Return all tracked files for Repo
+func (s *GlobSet) All() ([]File, error) {
+	files := make([]File, 0)
+	for _, pattern := range s.patterns {
+		matches, err := filepath.Glob(pattern)
+		if err != nil {
+			return files, err
+		}
+
+		for _, match := range matches {
+			if !filepath.IsAbs(match) {
+				match = filepath.Join(s.root, match)
+			}
+			matchRel, err := filepath.Rel(s.root, match)
+			if err != nil {
+				return files, err
+			}
```

Comment:
Why do we need the abs vs rel logic here?

Reply:
Both are used in snapshots
1. relative - we use as a remote name https://github.com/databricks/cli/blob/35e8ed30c6207c239331c30491ecd2c34b1d123e/libs/sync/snapshot.go#L217-L223

2. absolute - to load the file and work with it if needed (f.e. check if it's notebook https://github.com/databricks/cli/blob/35e8ed30c6207c239331c30491ecd2c34b1d123e/libs/sync/snapshot.go#L208)